spectator-web/src/main/java/com/netflix/spectator/controllers/model/TaggedDataPoints.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spectator.controllers.model;
+
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Measurement;
+import com.netflix.spectator.api.Tag;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A collection of DataPoint instances for a comment set of TagValues.
+ *
+ * This is only public for testing purposes so implements equals but not hash.
+ */
+public class TaggedDataPoints {
+  public static TaggedDataPoints make(Id id, List<Measurement> measurements,
+                                      boolean aggregate) {
+    List<TagValue> tags = new ArrayList<TagValue>();
+    List<DataPoint> dataPoints = new ArrayList<DataPoint>();
+    for (Tag tag : id.tags()) {
+        tags.add(new TagValue(tag.key(), tag.value()));
+    }
+    if (aggregate) {
+      DataPoint point = null;
+      for (Measurement measurement : measurements) {
+         if (point == null) {
+           point = DataPoint.make(measurement);
+           dataPoints.add(point);
+         } else {
+           point.aggregate(measurement);
```

Comment:
This might have been a bug in my test where my test Id wasnt implementing equals correctly so AggrMeter wasnt doing the aggregation. However I do think I had seen this in spinnaker so I'm going to TODO removing this after I have a chance to put it all together and verify. My tests suggest that I dont need it, but I might not have replicated the scenario I'd seen in spinnaker.